ctypes.go

```diff
@@ -124,3 +132,45 @@ func cstring(name string) *byte {
 	copy(b, name)
 	return &b[0]
 }
+
+// cast is used to centralized unsafe use C allocated pointer.
+// We take the address and then dereference it to trick go vet from creating a possible misuse of unsafe.Pointer
+func cast[T any](ptr uintptr) *T {
+	return (*T)(*(*unsafe.Pointer)(unsafe.Pointer(&ptr)))
+}
+
+// castWithOffset is the same as cast but adding an offset to the pointer by a multiple of the size
+// of the type pointed.
+func castWithOffset[T any](ptr uintptr, offset uint64) *T {
+	return (*T)(unsafe.Add(*(*unsafe.Pointer)(unsafe.Pointer(&ptr)), offset*uint64(unsafe.Sizeof(*new(T)))))
+}
+
+// ptrToUintptr is a helper to centralize of usage of unsafe.Pointer
+// do not use this function to cast interfaces
+func ptrToUintptr[T any](arg *T) uintptr {
+	return uintptr(unsafe.Pointer(arg))
+}
+
+func sliceToUintptr[T any](arg []T) uintptr {
+	return (*reflect.SliceHeader)(unsafe.Pointer(&arg)).Data
+}
+
+func stringToUintptr(arg string) uintptr {
+	return (*reflect.StringHeader)(unsafe.Pointer(&arg)).Data
+}
+
+var alwaysFalse bool
+var escapeSink any
+
+// keepAlive is a copy of runtime.KeepAlive
+// keepAlive has 2 usages:
+// It forces the deallocation of the memory to take place later than expected (just like runtime.KeepAlive), and
+// it forces the memory to be on the heap by calling a variadic parameter function which calls to go:linkname function.
+// Of course the if is always true and the function always returns but the compiler does not know this
```

Comment:
```suggestion
// - It forces the deallocation of the memory to take place later than expected (just like runtime.KeepAlive)
// - It forces the given argument x to be escaped on the heap by saving it into a global value (Go doesn't provide a standard way to do it as of today)
// It is implemented so that the compiler cannot optimize it.
```